.github/workflows/update-dockerfile.yaml

```diff
@@ -62,11 +68,12 @@ jobs:
             echo "No changes detected. Exiting workflow."
             echo "changes_detected=false" >> $GITHUB_OUTPUT
           else
-            echo "Changes detected, proceeding with commit."
+            echo "Changes detected, proceeding with commit if not a fork."
             echo "changes_detected=true" >> $GITHUB_OUTPUT
           fi
+
       - name: Commit and Push Changes
-        if: steps.check_changes.outputs.changes_detected == 'true'
+        if: steps.check_changes.outputs.changes_detected == 'true' && steps.repo_metadata.outputs.fork == 'false'
```

Comment:
I suggest to add this new check to the job, rather than this step (there is no value-add to run this job for forks at all)